app/boards/arm/adv360pro/adv360pro-layouts.dtsi

```diff
@@ -0,0 +1,90 @@
+#include <physical_layouts.dtsi>
+
+/{
+    default_layout: default_layout {
+        compatible = "zmk,physical-layout";
+        display-name = "Default";
+
+        transform = <&default_transform>;
+        kscan = <&kscan0>;
```

Comment:
I am really stuck on this aspect of the refactor. In particular, this makes this import still *highly* coupled with the importing environment, and makes it harder to refactor without surprising side effects. It also makes it so that future refactors that might identify "Hey, this layout is actually useful, and should be shared!" requires extracting this coupling later *anyways*.

It also leads to a "two ways to do things" situation, between "using shared layouts" and "using local layouts" that would potentially lead to more confusion. IMHO, I favor Joel's first proposed option.

Reply:
I think I'd like consistency between the shared and the local approaches on this point, which way around it goes I'm not too fussy about. That option is a bit ugly, but justifiable imo. What would your thoughts be on applying option 2 to all the shared layouts?

I *do* think that option 1 should be used for binding kscans to physical transforms when different kscans are used.

Reply:
IMHO matrix transforms are just as tightly coupled to the "host" that is going to inject them, and shouldn't be set either.